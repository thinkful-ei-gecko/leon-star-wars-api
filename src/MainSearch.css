
label {
  display: block;
  color: #d7bf36;
  font-weight: bold;
  margin-bottom: 5px;
}

select, option, input {
  background-color: gold;
  border: 1px solid white;
  color: black;
  margin-bottom: 15px;
  width: 100%;
  text-align: center;
  height: 20px;
}
