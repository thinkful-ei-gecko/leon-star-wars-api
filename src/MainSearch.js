import React from 'react'
import GoldDivider from './GoldDivider'
import SearchForm from './SearchForm'
import './MainSearch.css'
import './Button.css'
import ErrorBoundary from './ErrorBoundary'

export default function MainSearch(props) {
  return(
    <section>
    <p className="instructions">Enter a <span className="bold">search term</span> and <span className="bold">category</span> and we'll find what matches!</p>
    <GoldDivider />
    <ErrorBoundary>
      <SearchForm categories={props.categories} apiCall={props.apiCall} />
    </ErrorBoundary>
    </section>
  );
}