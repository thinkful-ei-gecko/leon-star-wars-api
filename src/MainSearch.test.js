import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MainSearch from './MainSearch'

let categoriesList = [ 'test', 'data', 'here', 'okay'];

describe('<MainSearch />', () => {

  it('renders empty fine', () => {
    const wrapper = shallow(<MainSearch />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
    
  it('renders with categories in select', () => {
    const wrapper = shallow(<MainSearch categories={categoriesList} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})