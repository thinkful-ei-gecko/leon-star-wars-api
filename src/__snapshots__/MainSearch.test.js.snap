// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainSearch /> renders empty fine 1`] = `
<section>
  <p
    className="instructions"
  >
    Enter a 
    <span
      className="bold"
    >
      search term
    </span>
     and 
    <span
      className="bold"
    >
      category
    </span>
     and we'll find what matches!
  </p>
  <GoldDivider />
  <ErrorBoundary>
    <SearchForm />
  </ErrorBoundary>
</section>
`;

exports[`<MainSearch /> renders with categories in select 1`] = `
<section>
  <p
    className="instructions"
  >
    Enter a 
    <span
      className="bold"
    >
      search term
    </span>
     and 
    <span
      className="bold"
    >
      category
    </span>
     and we'll find what matches!
  </p>
  <GoldDivider />
  <ErrorBoundary>
    <SearchForm
      categories={
        Array [
          "test",
          "data",
          "here",
          "okay",
        ]
      }
    />
  </ErrorBoundary>
</section>
`;
